src/Dali/RedSharp.Dali.Common.Interop/GlobalHotkeyProvider.cs

```diff
@@ -0,0 +1,264 @@
+        public bool Register(HotkeyModifier modifier, Key key)
+        {
+            if (IsDisposed)
+                throw new ObjectDisposedException(nameof(GlobalHotkeyProvider));
+
+            if (IsRegistered)
+                return false;
+
+            _holdedModifier = modifier;
+            _holdedKeyWinForm = (Keys)KeyInterop.VirtualKeyFromKey(key);
+            _holdedKeyWPF = key;
+
+            return InternalRegister();
+        }
+
+        /// <summary>
+        /// Registers new global HotKey combination.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">
+        /// Current instance has already disposed.
+        /// </exception>
+        /// <param name="key">Used WinForm key enum</param>
+        /// <returns>True if operation was success.</returns>
+        public bool Register(HotkeyModifier modifier, Keys key)
+        {
+            if (IsDisposed)
+                throw new ObjectDisposedException(nameof(GlobalHotkeyProvider));
+
+            if (IsRegistered)
+                return false;
+
+            _holdedModifier = modifier;
+            _holdedKeyWinForm = key;
+            _holdedKeyWPF = KeyInterop.KeyFromVirtualKey((int)key);
+
+            return InternalRegister();
+        }
+
+        /// <summary>
+        /// Remove already registered combination.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">
+        /// Current instance has already disposed.
+        /// </exception>
+        public void Unregister()
+        {
+            if (IsDisposed)
+                throw new ObjectDisposedException(nameof(GlobalHotkeyProvider));
+
+            if (!IsRegistered)
+                return;
+
+            NativeFunctions.UnregisterHotKey(_windowHandle, _identifier);
+
+            IsRegistered = false;
+        }
+
+        ~GlobalHotkeyProvider()
+        {
+            InternalDispose(false);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            InternalDispose(true);
+        }
+
+        #endregion
+
+        #region Private methods
+        //=================================================//
+        // Private methods
+
+        /// <summary>
+        /// Tried to register HotKey with current identifier.
+        /// </summary>
+        private bool InternalRegister()
+        {
+            bool result = NativeFunctions.RegisterHotKey(_windowHandle, _identifier, (int)_holdedModifier, (int)_holdedKeyWinForm);
+
+            if (!result)
+            {
+                Trace.WriteLine($"Can't register global HotKey with identifier: {_identifier}\n");
+
+                BannedIdentifiers.Add(_identifier);
+
+                Trace.WriteLine($"Identifier: {_identifier} was banned.\n");
+
+                _identifier = GetIdentifier();
+
+                Trace.WriteLine($"Provider get new identifier: {_identifier}\n");
+            }
+
+            IsRegistered = result;
+
+            return result;
```

Comment:
And in case of not registering, caller should call this method again?

Reply:
Yes.

Reply:
Maybe we should try again in public method?

Reply:
I delegate this option to the end user. Because we have a little chance to not register it in all attempts.